src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java

```diff
@@ -0,0 +1,102 @@
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.meta;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.hamcrest.MatcherAssert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.xmlunit.matchers.CompareMatcher;
+
+/**
+ * Test for {@link XmlStreamJoin}.
+ * @since 0.9
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public class XmlStreamJoinTest {
+
+    /**
+     * Test repo path.
+     */
+    private static final String REPO = "src/test/resources-binary/repodata";
+
+    @Test
+    void joinsTwoMetaXmlFiles(@TempDir final Path temp) throws IOException {
+        final Path file = Files.copy(
+            Paths.get(XmlStreamJoinTest.REPO, "primary.xml.example.first"),
+            temp.resolve("target.xml")
+        );
+        new XmlStreamJoin("metadata").merge(
+            file, Paths.get(XmlStreamJoinTest.REPO, "primary.xml.example.second")
+        );
+        MatcherAssert.assertThat(
+            file,
+            CompareMatcher.isIdenticalTo(Paths.get(XmlStreamJoinTest.REPO, "primary.xml.example"))
+        );
+    }
+
+    @Test
+    void joinsWhenXmlIsNotProperlyFormatted(@TempDir final Path temp) throws IOException {
+        final Path target = temp.resolve("res.xml");
+        final Path part = temp.resolve("part.xml");
+        Files.write(
+            target,
+            String.join(
+                "\n",
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+                "<parent><a>1</a>",
+                "<b>2</b></parent>"
+            ).getBytes()
+        );
+        Files.write(
+            part,
+            String.join(
+                "\n",
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+                "<parent><c>2</c>",
+                "<d>3</d></parent>"
+            ).getBytes()
+        );
+        new XmlStreamJoin("parent").merge(target, part);
+        final Path expected = temp.resolve("expected.xml");
+        Files.write(
+            expected,
+            String.join(
+                "\n",
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+                "<parent>",
+                "<a>1</a><b>2</b><c>2</c><d>3</d>",
+                "</parent>"
+            ).getBytes()
+        );
+        MatcherAssert.assertThat(
+            target,
+            CompareMatcher.isIdenticalTo(expected)
```

Comment:
@olenagerasimova Instead comparing if the two files are exactly the same, how about asserting if the generated file content xml is what we expect using xpath? I think that the test becomes more readable

Reply:
@paulodamaso but less reliable: we will never know if resulting file contains some extra tags/data

Reply:
@olenagerasimova Makes sense